app/services/katello/pulp3/repository.rb

```diff
@@ -109,11 +111,15 @@ def get_distribution(href)
         nil
       end
 
-      def delete_distributions_by_paths
+      def delete_distributions
         paths.values.each do |path|
-          dists = lookup_distributions(base_path: path.sub(/^\//, ''))
-          delete_distribution(dists.first._href) if dists.first
+          distribution_reference = distribution_reference(path)
+          if distribution_reference
+            delete_distribution(distribution_reference.href)
+            distribution_reference.destroy
+          end
         end
+        []
       end
```

Comment:
did you mean to be returning the tasks here? (looks like it'll always return empty array) :)

Reply:
I wasn't sure why but the delete distribution Pulp3 api task never returns a value (there is not result json, etc). I do see pulp/katello foreman logs showing he delete is successful, etc. In any case, I wasn't successful in returning an array of nil values, and this appeared to work (all test passed).

Reply:
ahhh, it may not be an async task, in which case we should make the Delete Distribution action a regular action and NOT an async action.  You might check the pulp3 api docs to confirm, also you might ask the devs on #pulp-dev, as the fact that create and update are async, but DELETE is almost seems like a bug to me (due to the reason they made create/update async)?

Reply:
According to the pulpcore docs it's not an async operation. Let's try it out.